### PR TITLE
Updated doc to reflect upgrade steps for release v2.10.0

### PR DESCRIPTION
**Pull request by red2k18**

### CODE REVIEW

File: _docs/_latest/setup/upgrade.md [hunk 1]
```diff
@@ -866,6 +866,152 @@ to the Client ID of your service account.
...
+bucket to your cloud shell (located under `forseti-server-xxxxxx/deployment_templates`) by running command  
+`gsutil cp gs://YOUR_FORSETI_GCS_BUCKET/deployment_templates/deploy-forseti-server-<LATEST_TEMPLATE>.yaml 
+deployment-templates/deploy-forseti-server-xxxxx-2-10-0.yaml`.
+1. Open up the deployment template `deployment-templates/deploy-forseti-server-xxxxx-2-10-0.yaml` for edit.
+    1. Update the `forseti-version` inside the deployment template to `tags/v2.10.0`.
+    
+1. Upload file `deployment-templates/deploy-forseti-server-xxxxx-2-10-0.yaml` back to the GCS bucket 
+(`forseti-server-xxxxxx/deployment_templates`) by running command  
+`gsutil cp deployment-templates/deploy-forseti-server-xxxxx-2-10-0.yaml gs://YOUR_FORSETI_GCS_BUCKET/
+deployment_templates/deploy-forseti-server-xxxxx-2-9-0.yaml`.
```
Comment (joecheuk) [inline _docs/_latest/setup/upgrade.md]: deploy-forseti-server-xxxxx-2-10-0.yaml

File: _docs/_latest/setup/upgrade.md [hunk 2]
```diff
@@ -866,6 +866,152 @@ to the Client ID of your service account.
...
+1. Navigate to [Deployment Manager](https://console.cloud.google.com/dm/deployments) and 
+copy the deployment name for Forseti server.
+1. Run command `gcloud deployment-manager deployments update DEPLOYMENT_NAME --config deploy-forseti-server-xxxxx-2-10-0.yaml`
+If you see errors while running the deployment manager update command, please refer to below section 
+`Error while running deployment manager` for details on how to workaround the error.
+1. Reset the Forseti server VM instance for changes in startup script to take effect.  
+You can reset the VM by running command `gcloud compute instances reset MY_FORSETI_SERVER_INSTANCE --zone MY_FORSETI_SERVER_ZONE`  
+Example command: `gcloud compute instances reset forseti-server-vm-70ce82f --zone us-central1-c`
+1. Repeat step `3-8` for Forseti client.
+1. To enable the [External Project Access Scanner]({% link _docs/latest/configure/scanner/descriptions.md %}#external-project-access-scanner), go to your Google Admin
```
Comment (joecheuk) [inline _docs/_latest/setup/upgrade.md]: I believe we didn't change the EPAS since release 2.9?

File: _docs/_latest/setup/upgrade.md [hunk 3]
```diff
@@ -866,6 +866,152 @@ to the Client ID of your service account.
...
+[Manage API client access](https://admin.google.com/ManageOauthClients) Security
```
Comment (joecheuk) [inline _docs/_latest/setup/upgrade.md]: same as above

File: _docs/_latest/setup/upgrade.md [hunk 4]
```diff
@@ -866,6 +866,152 @@ to the Client ID of your service account.
...
+settings and add API scope `https://www.googleapis.com/auth/cloudplatformprojects.readonly` 
+to the Client ID of your service account.
+    ```
+    https://www.googleapis.com/auth/admin.directory.group.readonly,https://www.googleapis.com/auth/admin.directory.user.readonly,https://www.googleapis.com/auth/cloudplatformprojects.readonly
+    ```
+1. Configuration file `forseti_conf_server.yaml` updates:
```
Comment (joecheuk) [inline _docs/_latest/setup/upgrade.md]: These are all 2.9 changes, can you update them with changes to 2.10?

File: _docs/_latest/setup/upgrade.md [hunk 5]
```diff
@@ -866,6 +866,71 @@ to the Client ID of your service account.
...
+        notifier:
+            email_connector:
+                name: sendgrid
+                auth:
+                  api_key: {SENDGRID_API_KEY}
+                sender: {EMAIL_SENDER}
+                recipient: {EMAIL_RECIPIENT}
+                data_format: csv
+        ```
+    - Update the `resources` section for all the resources to remove the
```
Comment (joecheuk) [inline _docs/_latest/setup/upgrade.md]: This is a big change, you should also mention that this is backward compatible if they decided not to do it.

File: _docs/_latest/setup/upgrade.md (hunk 4 above)
Comment (red2k18) [reply _docs/_latest/setup/upgrade.md]: Done

Comment (joecheuk) [reply _docs/_latest/setup/upgrade.md]: You still need to add this one back to inventory, api_timeout is a new key under inventory.

https://github.com/GoogleCloudPlatform/forseti-security/pull/2449/files

File: _docs/_latest/setup/upgrade.md [hunk 6]
```diff
@@ -866,6 +866,71 @@ to the Client ID of your service account.
 {% endcapture %}
 {% include site/zippy/item.html title="Upgrading 2.8.0 to 2.9.0" content=upgrading_2_8_0_to_2_9_0 uid=10 %}
 
+{% capture upgrading_2_9_0_to_2_10_0 %}
+
+1. Open cloud shell when you are in the Forseti project on GCP.
```
Comment (joecheuk) [inline _docs/_latest/setup/upgrade.md]: You need to add a section for role updates:

https://github.com/GoogleCloudPlatform/forseti-security/pull/2395

File: _docs/_latest/setup/upgrade.md (hunk 5 above)
Comment (red2k18) [reply _docs/_latest/setup/upgrade.md]: Consider it done

Comment (red2k18) [reply _docs/_latest/setup/upgrade.md]: Done

File: _docs/_latest/setup/upgrade.md (hunk 4 above)
Comment (red2k18) [reply _docs/_latest/setup/upgrade.md]: Done

File: _docs/_latest/setup/upgrade.md (hunk 2 above)
Comment (red2k18) [reply _docs/_latest/setup/upgrade.md]: Yes, we haven't. Removed it.

File: _docs/_latest/setup/upgrade.md [hunk 7]
```diff
@@ -866,6 +866,85 @@ to the Client ID of your service account.
...
+    - Update the `api_timeout` field to configure Cloud Asset API timeout. 
+        ```
+        inventory:
+            ...
+            cai:
+                # Timeout in seconds to wait for the exportAssets API to return success.
+                # Defaults to 3600 if not set.
+                api_timeout: 3600
+        ``` 
+    **Notifier**
```
Comment (blueandgold) [inline _docs/_latest/setup/upgrade.md]: If the user is making this change here, they should also remove the section from the global config.

File: _docs/_latest/setup/upgrade.md [hunk 8]
```diff
@@ -866,6 +866,93 @@ to the Client ID of your service account.
...
+                        configuration:
+                        data_format: csv
+                        # gcs_path should begin with "gs://"
+                        gcs_path: gs://{FORSETI_BUCKET}/scanner_violations  
+        ```
+    - If you are making this change, please remove the email configuration 
+    from `global` section.
+    ```
+    global:
+        email_recipient: {EMAIL_RECIPIENT}
```
Comment (blueandgold) [inline _docs/_latest/setup/upgrade.md]: Please also tell the user to add the placeholder value.

File: _docs/_latest/setup/upgrade.md [hunk 9]
```diff
@@ -866,6 +866,98 @@ to the Client ID of your service account.
...
+`Error while running deployment manager` for details on how to workaround the error.
+1. Reset the Forseti server VM instance for changes in startup script to take effect.  
+You can reset the VM by running command `gcloud compute instances reset MY_FORSETI_SERVER_INSTANCE --zone MY_FORSETI_SERVER_ZONE`  
+Example command: `gcloud compute instances reset forseti-server-vm-70ce82f --zone us-central1-c`
+1. Repeat step `3-8` for Forseti client.
+1. Configuration file `forseti_conf_server.yaml` updates: 
+    **Inventory**
+    - Update the `api_timeout` field to configure Cloud Asset API timeout. 
+        ```
+        inventory:
```
Comment (blueandgold) [inline _docs/_latest/setup/upgrade.md]: You also need to add the new asset types from here.
image

File: _docs/_latest/setup/upgrade.md [hunk 10]
```diff
@@ -866,6 +866,113 @@ to the Client ID of your service account.
...
+            ...
+            cai:
+                # Timeout in seconds to wait for the exportAssets API to return success.
+                # Defaults to 3600 if not set.
+                api_timeout: 3600
+        ``` 
+    - Update the `cai` section to include the following newly fetched asset
```
Comment (blueandgold) [inline _docs/_latest/setup/upgrade.md]: You should combine the cai timeout and the new asset types here.
```
Update the `cai` section to include the new `api_timeout` field and the newly fetched asset:

inventory
    ...
    cai:
        # Timeout in seconds to wait for the exportAssets API to return success.
        # Defaults to 3600 if not set.
        api_timeout: 3600

        # If commented out then all currently supported asset types are
        # exported from Cloud Asset API. The list of default asset types is
        # in google/cloud/forseti/services/inventory/base/cloudasset.py
               
        #asset_types:
        #   - google.cloud.sql.Instance
        #   - google.compute.VpnTunnel
        #   - google.pubsub.Subscription
```

**Pull request merged**